code/modules/surgery/organs_internal.dm

```diff
@@ -1,432 +1,527 @@
-			for(var/datum/reagent/consumable/ethanol/alcohol in R.reagent_list)
-				ethanol += alcohol.alcohol_perc * 300
-			ethanol /= C.reagents.reagent_list.len
 
-		for(var/obj/item/organ/internal/I in affected.internal_organs)
-			I.germ_level = max(I.germ_level-ethanol, 0)
-			I.take_damage(rand(4,8),0)
+/datum/surgery_step/internal/manipulate_organs/insert/mmi/can_use(mob/living/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/active_surgery/surgery)
+	var/obj/item/device/mmi/M = tool
 
-		R.trans_to(target, GHETTO_DISINFECT_AMOUNT * 10)
-		R.reaction(target, INGEST)
+	if(!ishuman(target))
+		return 0
+	var/mob/living/carbon/human/H = target
+	affected = H.get_organ(target_zone)
 
-		user.visible_message("<span class='warning'> [user]'s hand slips, splashing the contents of [tool] all over [target]'s [affected.name] incision!</span>", \
-		"<span class='warning'> Your hand slips, splashing the contents of [tool] all over [target]'s [affected.name] incision!</span>")
+	if(!istype(M))
 		return 0
 
-	else if(current_type == "extract")
-		if(I && I.owner == target)
-			if(affected)
-				user.visible_message("<span class='warning'> [user]'s hand slips, damaging [target]'s [affected.name] with [tool]!</span>", \
-				"<span class='warning'> Your hand slips, damaging [target]'s [affected.name] with [tool]!</span>")
-				affected.createwound(BRUISE, 20)
-			else
-				user.visible_message("<span class='warning'> [user]'s hand slips, damaging [target]'s [parse_zone(target_zone)] with [tool]!</span>", \
-				"<span class='warning'> Your hand slips, damaging [target]'s [parse_zone(target_zone)] with [tool]!</span>")
-		else
-			user.visible_message("[user] can't seem to extract anything from [target]'s [parse_zone(target_zone)]!",
-				"<span class='notice'>You can't extract anything from [target]'s [parse_zone(target_zone)]!</span>")
+	if(!M.brainmob || !M.brainmob.client || !M.brainmob.ckey || M.brainmob.stat >= DEAD)
+		user << "<span class='danger'>That brain is not usable.</span>"
 		return 0
 
-	else if(current_type == "finish")
-		if(affected && affected.encased)
-			var/msg = "<span class='warning'> [user]'s hand slips, bending [target]'s [affected.encased] the wrong way!</span>"
-			var/self_msg = "<span class='warning'> Your hand slips, bending [target]'s [affected.encased] the wrong way!</span>"
-			user.visible_message(msg, self_msg)
-			affected.fracture()
-		else
-			var/msg = "<span class='warning'> [user]'s hand slips, tearing the skin!</span>"
-			var/self_msg = "<span class='warning'> Your hand slips, tearing skin!</span>"
-			user.visible_message(msg, self_msg)
-		if(affected)
-			affected.createwound(BRUISE, 20)
+	if(!(affected && affected.status & ORGAN_ROBOT))
+		user << "<span class='danger'>You cannot install a computer brain into a meat enclosure.</span>"
 		return 0
 
+	if(!H.species.has_organ["brain"])
+		user << "<span class='danger'>You're pretty sure [H.species.name_plural] don't normally have a brain.</span>"
```

Comment:
`user <<`

`don't` -> `doesn't` - grammatically, maybe? Can someone else confirm? My brain is feeling a bit fried.